backend/director/llm/mistral.py

```diff
@@ -0,0 +1,205 @@
+from enum import Enum
+
+from pydantic import Field, field_validator, FieldValidationInfo
+from pydantic_settings import SettingsConfigDict
+
+from director.core.session import RoleTypes
+from director.llm.base import BaseLLM, BaseLLMConfig, LLMResponse, LLMResponseStatus
+from director.constants import (
+    LLMType,
+    EnvPrefix,
+)
+import json
+
+
+class MistralChatModel(str, Enum):
+    """Enum for Mistral Chat models"""
+
+    MINISTRAL_8B = "ministral-8b-latest"
+    MINISTRAL_3B = "ministral-3b-latest"
+    # PIXTRAL_12B = "pixtral-12b-2409"
+  
+
+
+class MistralAIConfig(BaseLLMConfig):
+    """MistralAI Config"""
+
+    model_config = SettingsConfigDict(
+        env_prefix=EnvPrefix.MISTRAL_,
+        extra="ignore",
+    )
+
+    llm_type: str = LLMType.MISTRAL
+    api_key: str = ""
+    api_base: str = ""
+    chat_model: str = Field(default=MistralChatModel.MINISTRAL_8B)
+
+    @field_validator("api_key")
+    @classmethod
+    def validate_non_empty(cls, v, info: FieldValidationInfo):
+        if not v:
+            raise ValueError(
+                f"{info.field_name} must not be empty. please set {EnvPrefix.MISTRAL_.value}{info.field_name.upper()} environment variable."
+            )
+        return v
+
+
+class MistralAI(BaseLLM):
+    def __init__(self, config: MistralAIConfig = None):
+        """
+        :param config: MistralAI Config
+        """
+        if config is None:
+            config = MistralAIConfig()
+        super().__init__(config=config)
+        try:
+            import mistralai
```

Comment:
Please add the `mistralai` library to the list of dependencies in the requirements.txt file